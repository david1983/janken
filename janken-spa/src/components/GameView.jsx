import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import {
  FormControl, MenuItem, Button, InputLabel, Select,
} from '@material-ui/core';

import GameClass from '../state/Game';

/**
 * Component that renders the main game view
 * where the players can choose their moves
 * @param {props} props The component props
 */
const GameView = ({ Game, playerNumber }) => {
  const player = Game.getPlayer(playerNumber);
  return (
    <div className="game-view-content">

      <h3>
        {`${[player.name]}'s turn`}
      </h3>

      {player.move === '' && (
        <div>

          <FormControl className="form-control-select-move">

            <InputLabel htmlFor="move-simple">Choose a move</InputLabel>
            <Select
              inputProps={{
                name: 'move',
                id: 'move-simple',
              }}
              value="Chose a move"
              onChange={e => Game.doMove(playerNumber, e.target.value)}
            >
              {Game.uniqueMoves.map(move => (
                <MenuItem key={`select-move-${move}`} value={move}>{move}</MenuItem>
              ))}

            </Select>
          </FormControl>

        </div>
      )
      }

      {player.move !== '' && (
        <div>
          <div>
            {`${player.name}'s choice is ${player.move}`}
          </div>
          <hr />
          <Button onClick={() => Game.doMove(playerNumber, '')}>Change move</Button>
        </div>
      )}

    </div>
  );
};
GameView.defaultProps = {
  Game: new GameClass(),
  playerNumber: 1,
};

GameView.propTypes = {
  Game: PropTypes.object,
  playerNumber: PropTypes.number,
};


export default inject('Game')(observer(GameView));
