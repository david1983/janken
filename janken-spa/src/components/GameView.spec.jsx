import React from 'react';
import { render } from 'enzyme';
import GameView from './GameView';
import GameClass from '../state/Game';

const Game = new GameClass();
describe('the GameView component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <GameView Game={Game} />;
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
  });
});
