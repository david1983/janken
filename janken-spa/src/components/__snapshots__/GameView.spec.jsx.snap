// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GameView component should match the snapshot of the shallow rendered component 1`] = `
<div
  class="game-view-content"
>
  <h3>
    's turn
  </h3>
  <div>
    <div
      class="MuiFormControl-root form-control-select-move"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink="true"
        for="move-simple"
      >
        Choose a move
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <div
          class="MuiSelect-root"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="select-move"
            role="button"
            tabindex="0"
          />
          <input
            id="move-simple"
            name="move"
            type="hidden"
            value="Chose a move"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
